sycl/ur_win_proxy_loader/ur_win_proxy_loader.cpp

```diff
@@ -158,63 +156,29 @@ void preloadLibraries() {
 
 /// windows_pi.cpp:loadOsPluginLibrary() calls this to get the DLL loaded
 /// earlier.
-__declspec(dllexport) void *getPreloadedPlugin(
-    const std::filesystem::path &PluginPath) {
-
-  MapT &dllMap = getDllMap();
-
-  // All entries in the dllMap have the same parent directory.
-  // To avoid case sensivity issues, we don't want to do string comparison but
-  // just make sure that directory of the entires in the map and directory of
-  // the PluginPath are equivalent (point to the same physical location).
-  auto match = dllMap.end();
-  std::error_code ec;
-  if (!dllMap.empty() &&
-      std::filesystem::equivalent((dllMap.begin())->first.parent_path(),
-                                  PluginPath.parent_path(), ec)) {
-    // Now we can search only by filename. Result might be nullptr (not found),
-    // which is perfectly valid.
-    match =
-        std::find_if(dllMap.begin(), dllMap.end(),
-                     [&](const std::pair<std::filesystem::path, void *> &v) {
-                       return v.first.filename() == PluginPath.filename();
-                     });
-  }
-
-  if (match == dllMap.end()) {
-    // unit testing? return nullptr (not found) rather than risk asserting below
-    if (PluginPath.string().find("unittests") != std::string::npos)
-      return nullptr;
-
-    // Otherwise, asking for something we don't know about at all, is an issue.
-    std::cout << "unknown plugin: " << PluginPath << std::endl;
-    assert(false && "getPreloadedPlugin was given an unknown plugin path.");
-    return nullptr;
-  }
-  return match->second;
-}
+__declspec(dllexport) void *getPreloadedURLib() { return getDllHandle(); }
 
 BOOL WINAPI DllMain(HINSTANCE hinstDLL, // handle to DLL module
                     DWORD fdwReason,    // reason for calling function
                     LPVOID lpReserved)  // reserved
 {
-  bool PrintPiTrace = false;
-  static const char *PiTrace = std::getenv("SYCL_PI_TRACE");
-  static const int PiTraceValue = PiTrace ? std::stoi(PiTrace) : 0;
-  if (PiTraceValue == -1 || PiTraceValue == 2) { // Means print all PI traces
-    PrintPiTrace = true;
+  bool PrintUrTrace = false;
+  static const char *UrTrace = std::getenv("SYCL_UR_TRACE");
+  static const int UrTraceValue = UrTrace ? std::stoi(UrTrace) : 0;
```

Comment:
a try catch might be needed to avoid crashing when `SYCL_UR_TRACE` is malformed.

Reply:
I've addressed this now, although it will still crash elsewhere until https://github.com/oneapi-src/unified-runtime/pull/2059 lands